cmd/routers.go

```diff
@@ -120,5 +121,20 @@ func configureServerHandler(endpoints EndpointList) (http.Handler, error) {
 	registerAPIRouter(router)
 
 	// Register rest of the handlers.
-	return registerHandlers(router, globalHandlers...), nil
+	return criticalErrorHandler{registerHandlers(router, globalHandlers...)}, nil
+}
+
+// criticalErrorHandler handles critical server failures caused by
+// `panic(logger.ErrCritical)` as done by `logger.CriticalIf`.
+type criticalErrorHandler struct{ handler http.Handler }
+
+func (h criticalErrorHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	defer func() {
+		if err := recover(); err == logger.ErrCritical { // handle
+			w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Perhaps we could write an XML response here since we have the control.

Reply:
Actually I wasn't happy with my solution - moved handler to `cmd/http` (One central place).
But I can manually write XML response if requested?

Reply:
Hm, actually we cannot write proper XML response from `cmd/server`. Is it important that client get's S3 XML response in case of critical failure. In my understanding `logger.CriticalIf` is only for logical / irrecoverable errors - like an assumed invariant is broken...

Reply:
It is important to write an XML response otherwise the clients which see this is going to error out failing to parse the XML instead of an actual error. It will be unexpected for an S3 client.

Reply:
Hm, okay... Than I have to implement it in `cmd` and pass it to `http.NewServer`...